src/main/java/org/commonjava/storage/pathmapped/datastax/CassandraPathDB.java

```diff
@@ -230,23 +230,30 @@ public void insert( PathMap pathMap, String checksum )
             delete( fileSystem, path );
         }
 
-        final FileChecksum existedChecksum = fileChecksumMapper.get( checksum );
-
-        if ( existedChecksum != null )
+        //TODO: This checksum null checking is used to add a workaround for storage migration. We are judging if
+        //      this checksum entry is needed for existed legacy storage files, so if not needed, null can be passed
+        //      in to skip the checksum de-dupe process.
+        if ( checksum != null && !"".equals( checksum.trim() ) )
```

Comment:
use something like isNotBlank()?

Reply:
Seems we have direct dep on commons-lang which I think we don't have before. Ok I'll use it.